Framework/DataHandling/src/SaveOpenGenieAscii.cpp

```diff
@@ -49,337 +46,382 @@ void SaveOpenGenieAscii::init() {
   std::vector<std::string> header{"ENGIN-X Format", "Basic Format"};
   declareProperty("OpenGenieFormat", "ENGIN-X Format",
                   boost::make_shared<Kernel::StringListValidator>(header),
-                  "The format required to succesfully load the file to "
+                  "The format required to successfully load the file to "
                   "OpenGnie: ENGIN-X Format (default)");
-  declareProperty(
-      "SpecNumberField", "",
-      "Spec number is a required field for ENGIN-X format of OpenGenie,"
-      "an example of spec number would be: 1201 - 1400");
-  setPropertySettings("SpecNumberField",
-                      make_unique<VisibleWhenProperty>(
-                          "OpenGenieFormat", IS_EQUAL_TO, "ENGIN-X Format"));
 }
 
+/**
+  * Writes an OpenGenie file in ASCII at the user specified path
+  */
 void SaveOpenGenieAscii::exec() {
-  // Process properties
-
   // Retrieve the input workspace
-  /// Workspace
-  ws = getProperty("InputWorkspace");
-  int nSpectra = static_cast<int>(ws->getNumberHistograms());
-  int nBins = static_cast<int>(ws->blocksize());
-
-  // Retrieve the filename from the properties
-  std::string filename = getProperty("Filename");
+  m_inputWS = getProperty("InputWorkspace");
 
-  // Output string variables
-  const std::string singleSpc = " ";
-  const std::string fourspc = "    ";
+  inputValidation();
 
-  // file
-  std::ofstream outfile(filename.c_str());
-  if (!outfile) {
-    g_log.error("Unable to create file: " + filename);
-    throw Exception::FileError("Unable to create file: ", filename);
-  }
-  if (nBins == 0 || nSpectra == 0)
-    throw std::runtime_error("Trying to save an empty workspace");
+  // Its better to reserve over and the number of logs gives us a good
+  // estimate as some logs are not included whilst some other params are.
+  m_outputVector.reserve(m_inputWS->run().getLogData().size());
 
-  // Axis alphabets
-  const std::string Alpha[] = {"\"e\"", "\"x\"", "\"y\""};
+  // Whilst this doesn't weigh in the processing
+  // required it breaks down the algorithm nicely for the moment
+  const int numOfSteps = 6;
+  Progress progressBar(this, 0, 1, numOfSteps);
 
-  // Get IncludeHeader property
-  const bool headers = getProperty("IncludeHeader");
-  // if true write file header
-  if (headers) {
-    writeFileHeader(outfile);
+  const std::string formatType = getProperty("OpenGenieFormat");
+  if (formatType == "ENGIN-X Format") {
+    progressBar.report("Generating ENGINX header");
+    applyEnginxFormat();
+  } else {
+    const std::string err_msg("Unrecognized format \"" + formatType + "\"");
+    throw std::runtime_error(err_msg);
```

Comment:
I think the smartest option would be to remove the "Basic Format" option. This is because the algorithm is written for ENGIN-X OpenGenie which is the normal package plus several bespoke routines.
I suspect the original writer(s) foresaw the possibility of needing to support the vanilla version of the package and tried to keep the ENGIN-X specific stuff tided.